File: Frontend/вчараырапрвапр/model/Product.js
/**
 * Модель товара в каталоге
 */
Ext.define('Marketplace.model.Product', {
    extend: 'Ext.data.Model',

    fields: [
        { name: 'id', type: 'int' },
        { name: 'name', type: 'string' },
        { name: 'brand', type: 'string' },
        { name: 'category', type: 'string' },
        { name: 'price', type: 'float' },
        { name: 'oldPrice', type: 'float' },
        { name: 'image', type: 'string' },
        { name: 'rating', type: 'float' },
        { name: 'reviewsCount', type: 'int' },
        { name: 'weight', type: 'string' },
        { name: 'description', type: 'string' },
        { name: 'inStock', type: 'boolean' },
        { name: 'stockQuantity', type: 'int' },
        { name: 'isFavorite', type: 'boolean' },
        { name: 'inCart', type: 'boolean' }
    ],

    validators: {
        name: { type: 'presence', message: 'Название товара обязательно' },
        price: { type: 'presence', message: 'Цена обязательна' }
    },

    /**
     * Проверка наличия скидки
     * @returns {Boolean}
     */
    hasDiscount: function() {
        return this.get('oldPrice') && this.get('oldPrice') > this.get('price');
    },

    /**
     * Размер скидки в процентах
     * @returns {Number}
     */
    getDiscountPercent: function() {
        if (!this.hasDiscount()) return 0;
        return Math.round((1 - this.get('price') / this.get('oldPrice')) * 100);
    }
});

File: Backend/Models/Product.cs
using System.ComponentModel.DataAnnotations;

namespace net_chuvgu.Backend.Models;

public class Product
{
    public int Id { get; set; }
        
    [Required]
    [StringLength(100)]
    public string Name { get; set; }
        
    [Required]
    [StringLength(50)]
    public string Brand { get; set; }
        
    [Required]
    [StringLength(50)]
    public string Category { get; set; }
        
    [Range(0, 100000)]
    public decimal Price { get; set; }
        
    [Range(0, 100000)]
    public decimal? OldPrice { get; set; }
        
    public string Image { get; set; }
        
    [Range(0, 5)]
    public double Rating { get; set; }
        
    public int ReviewsCount { get; set; }
    public string Weight { get; set; }
    public string Description { get; set; }
    public bool InStock { get; set; }
    public int StockQuantity { get; set; }
    public DateTime CreatedAt { get; set; }
        
    // ИЗМЕНИ НА ЭТО:
    public virtual ICollection<UserFavorite> UserFavorites { get; set; } = new List<UserFavorite>();
    public virtual ICollection<CartItem> CartItems { get; set; } = new List<CartItem>();
    public virtual ICollection<OrderItem> OrderItems { get; set; } = new List<OrderItem>();
}
